File: ContactsManagerAppWithJavaROOMDBMVVM/app/src/main/java/com/masterclass/contactsmanagerappwithjavaroomdbmvvm/viewModels/MyViewModel.java
package com.masterclass.contactsmanagerappwithjavaroomdbmvvm.viewModels;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

import com.masterclass.contactsmanagerappwithjavaroomdbmvvm.database.Contacts;
import com.masterclass.contactsmanagerappwithjavaroomdbmvvm.repositories.ContactsRepository;

import java.util.List;

public class MyViewModel extends AndroidViewModel {

    private ContactsRepository repository;
    private LiveData<List<Contacts>> allContacts;

    public MyViewModel(@NonNull Application application) {
        super(application);
        this.repository = new ContactsRepository(application);
    }

    public LiveData<List<Contacts>> getAllContacts(){
        allContacts = repository.getAllContacts();
        return allContacts;
    }

    public void addNewContact(Contacts contact){
        repository.addContact(contact);
    }

    public void deleteContact(Contacts contact){
        repository.deleteContact(contact);
    }
}

File: ContactsManagerAppWithKotlinROOMDBMVVM2/app/src/main/java/com/masterclass/contactsmanagerappwithkotlinroomdbmvvm/viewModels/MyViewModel.kt
package com.masterclass.contactsmanagerappwithkotlinroomdbmvvm.viewModels

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import com.masterclass.contactsmanagerappwithkotlinroomdbmvvm.database.Contacts
import com.masterclass.contactsmanagerappwithkotlinroomdbmvvm.repositories.ContactsRepository

class MyViewModel(application: Application) : AndroidViewModel(application) {

    private val repository: ContactsRepository = ContactsRepository(application)
    private var allContacts: LiveData<List<Contacts>>? = null

    init {
        allContacts = repository.getAllContacts()
    }

    fun getAllContacts(): LiveData<List<Contacts>> {
        return allContacts!!
    }

    fun addNewContact(contact: Contacts) {
        repository.addContact(contact)
    }

    fun deleteContact(contact: Contacts) {
        repository.deleteContact(contact)
    }
}